client/app/bundles/course/assessment/components/AssessmentForm/index.tsx

```diff
@@ -50,13 +50,18 @@ const AssessmentForm = (props: AssessmentFormProps): JSX.Element => {
     monitoringEnabled,
   } = props;
 
+  const defaultValues = {
+    ...initialValues,
+    has_time_limit: !!initialValues.time_limit,
```

Comment:
Why not add this where initialValues object is first created?

Reply:
moved the definition to the component several steps above it